nasl/nasl_host.c

```diff
@@ -234,6 +235,51 @@ get_host_open_port (lex_ctxt *lexic)
   return retc;
 }
 
+tree_cell *
+host_reverse_lookup (lex_ctxt *lexic)
+{
+  char *t = get_str_var_by_num (lexic, 0);
+  gvm_host_t *target = NULL;
+  tree_cell *retc;
+
+  if (t == NULL)
+    {
+      t = plug_get_host_ip_str (lexic->script_infos);
+    }
+  else
+    {
+      // we need to duplicate it as get_str_var_by_name does store it within
+      // string_form which is released with the lex_ctxt release.
+      t = g_strdup (t);
+    }
+  if (t == NULL)
+    {
+      nasl_perror (lexic, "Empty target\n");
+      goto fail;
+    }
+  target = gvm_host_from_str (t);
+  if (target == NULL)
+    {
+      nasl_perror (lexic, "%s: Invalid target\n", t);
+      goto fail;
```

Comment:
No, `t` comes from `get_str_ver_by_num` that returns the pointer from lex_ctxt and is freed when the lexer is freed. Therefore it would run into a double_free when I free t.

Reply:
you're right, t gets duplicated. My bad.